### PR TITLE
OpenStackLatentWorker: Renderable fixes

### PR DESCRIPTION
**Pull request by cmouse**
#5041  Contributor Checklist:

* [x] I have updated the unit tests
* [x] I have created a file in the `master/buildbot/newsfragments` directory (and read the `README.txt` in that directory)
* [x] I have updated the appropriate documentation

### CODE REVIEW

File: master/buildbot/worker/openstack.py
Comment (p12tic) [inline master/buildbot/worker/openstack.py:247]: Since this is private, let's name it as `_get_flavor`.

File: master/docs/manual/configuration/workers-openstack.rst
Comment (p12tic) [inline master/docs/manual/configuration/workers-openstack.rst:56]: It's unclear what type of the flavor is.

`It will be passed the list of available flavors as a list of strings and must return the flavor to use as string.`

File: master/buildbot/worker/openstack.py
Comment (cmouse) [reply master/buildbot/worker/openstack.py:247]: This is not consistent with the rest of the class, I can rename it but it will be different from others.

File: master/buildbot/worker/openstack.py
Comment (seankelly) [inline master/buildbot/worker/openstack.py:212]: Probably safe to delete this. Nova removed this part a long time ago (novaclient 8.0 I think?)

File: master/buildbot/worker/openstack.py
Comment (seankelly) [inline master/buildbot/worker/openstack.py:251]: I know I used `callable` for the `images` method but I think only using a renderer fits better with the rest of Buildbot.

File: master/buildbot/worker/openstack.py
Comment (seankelly) [reply master/buildbot/worker/openstack.py:247]: My recollection is Buildbot follows Twisted's style and that would be `_getFlavor`.

File: master/buildbot/worker/openstack.py
Comment (cmouse) [reply master/buildbot/worker/openstack.py:251]: it's actually quite useful, because otherwise you cannot easily resolve `m1.medium` into UUID.

Comment (cmouse) [reply master/buildbot/worker/openstack.py:251]: Example code:
```.py
# Used by openstack client - return matching flavor uuid
def os_get_flavor(flavors):
    want_flavor = util.Interpolate('%(prop:os_flavos)s')
    for flavor in flavors:
        if flavor.name == want_flavor:
            return flavor.id
    return want_flavor

# Used by openstack client - return matching image uuid
def os_get_image(images):
    want_image = util.Interpolate('%(prop:os_image)s')
    for image in images:
        if image.name == want_image:
            return image.id
    return want_image
```

File: master/buildbot/worker/openstack.py
Comment (cmouse) [reply master/buildbot/worker/openstack.py:212]: I feel bit bad about removing it. I know it's ages ago, but then again, it's very small tweak to maintain compability with ancient openstack.

File: master/buildbot/worker/openstack.py
Comment (cmouse) [reply master/buildbot/worker/openstack.py:251]: Actually, maybe we could just move this code inside the worker? Would that be ok?

Comment (seankelly) [reply master/buildbot/worker/openstack.py:251]: I would definitely prefer to be able to use the name instead of the UUID so I think it belongs in Buildbot so others can also benefit.

File: master/buildbot/worker/openstack.py
Comment (p12tic) [reply master/buildbot/worker/openstack.py:238]: We've changed this recently, see http://docs.buildbot.net/latest/developer/style.html

Comment (cmouse) [reply master/buildbot/worker/openstack.py:238]: @p12tic so what should be done? Fix the entire file, just this instance or?

**Review by p12tic — APPROVED** [review]

**Comment by p12tic** [comment]
I'll wait for @seankelly approval as there were further comments.

File: master/buildbot/worker/openstack.py
Comment (p12tic) [reply master/buildbot/worker/openstack.py:238]: I think we can use the new approach on the newly added code only. This does introduce inconsistency, but as the time goes, more and more of Buildbot will migrate and this will become less of an issue.

**Review by seankelly — APPROVED** [review]

**Pull request merged**
